x/epoching/keeper/keeper.go

```diff
@@ -0,0 +1,189 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	db "github.com/tendermint/tm-db"
+)
+
+// keys
+var (
+	NextEpochActionID      = []byte("next_epoch_action_id")
+	EpochNumberID          = []byte{0x50}
+	EpochActionQueuePrefix = []byte{0x51} // prefix for the epoch
+)
+
+// Keeper of the store
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      codec.BinaryCodec
+}
+
+// NewKeeper creates a epoch queue manager
+func NewKeeper(cdc codec.BinaryCodec, key sdk.StoreKey) Keeper {
+
+	return Keeper{
+		storeKey: key,
+		cdc:      cdc,
+	}
+}
+
+// GetNewActionID returns ID to be used for next epoch
+func (k Keeper) GetNewActionID(ctx sdk.Context) uint64 {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(NextEpochActionID)
+	if bz == nil {
+		// return default action ID to 1
+		return 1
```

Comment:
nit: can we store this in a `const`?

```suggestion
		return DefaultEpochActionID
```